plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteClient.java

```diff
@@ -0,0 +1,594 @@
+                        Optional.of(decimalType.getPrecision()),
+                        Optional.of(decimalType.getScale()),
+                        Optional.empty(),
+                        Optional.empty()));
+    }
+
+    public static ColumnMapping varcharColumnMapping(Optional<Integer> columnSize)
+    {
+        VarcharType varcharType = columnSize
+                // default added column in Ignite will has Integer.MAX_VALUE length which not allow in Trino, will consider the column as varchar
+                .filter(size -> size != UNBOUNDED_LENGTH)
+                .map(VarcharType::createVarcharType)
+                .orElse(createUnboundedVarcharType());
+        return ColumnMapping.sliceMapping(varcharType, varcharReadFunction(varcharType), varcharWriteFunction(), FULL_PUSHDOWN);
+    }
+
+    private static ColumnMapping charColumnMapping(Optional<Integer> charLength)
+    {
+        if (charLength.isEmpty() || charLength.get() > CharType.MAX_LENGTH) {
+            return varcharColumnMapping(charLength);
+        }
+        CharType charType = createCharType(charLength.get());
+        return ColumnMapping.sliceMapping(
+                charType,
+                charReadFunction(charType),
+                charWriteFunction());
+    }
+
+    public static LongReadFunction dateReadFunction()
+    {
+        return (resultSet, columnIndex) -> {
+            long localMillis = resultSet.getDate(columnIndex).getTime();
+            return MILLISECONDS.toDays(localMillis);
+        };
+    }
+
+    public static LongWriteFunction dateWriteFunction()
+    {
+        return (statement, index, value) -> {
+            long millis = DAYS.toMillis(value);
+            statement.setDate(index, new Date(millis));
+        };
+    }
+
+    @Override
+    public JdbcOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        if (tableMetadata.getComment().isPresent()) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating tables with table comment");
+        }
+
+        SchemaTableName schemaTableName = tableMetadata.getTable();
+        String schema = schemaTableName.getSchemaName();
+        if (!getSchemaNames(session).contains(schema)) {
+            throw new SchemaNotFoundException(schema);
+        }
+
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            List<String> columns = tableMetadata.getColumns().stream()
+                    .map(col -> getColumnDefinitionSql(session, col, col.getName()))
+                    .collect(toImmutableList());
+            List<String> columnNames = tableMetadata.getColumns().stream()
+                    .map(ColumnMetadata::getName)
+                    .collect(toImmutableList());
+            List<Type> columnTypes = tableMetadata.getColumns().stream()
+                    .map(ColumnMetadata::getType)
+                    .collect(toImmutableList());
```

Comment:
nit: Iterating all columns three times isn't nice when it takes many columns. It would be nice to build these 3 objects in a loop.